### DISCUSSION TITLE
Can golang AWS s3 client work with seaweedfs?

### DISCUSSION THREAD

**Discussion by palandlom** [seaweedfs/seaweedfs#4578] [Q&A] [answered]
We use `github.com/aws/aws-sdk-go-v2` module in our golang-app to interact with AWS S3 service. And now we consider switching to seaweedfs S3 storage.

Can we use `aws-sdk-go-v2` for interaction with `seaweedfs` service which is started in s3 mode i.e. 
```bash
weed -v 3 server -dir=./data -volume.max=50 -s3 -s3.config=config.json 
```
... or it is not possible and we need to use [goseaweedfs-client](https://github.com/linxGnu/goseaweedfs/) instead?

**Comment by chrislusf** [comment, marked as answer]
it should

**Comment by brawler2011** [comment]
Could someone provide an example?

**Reply by palandlom** [reply]
see https://github.com/seaweedfs/seaweedfs/discussions/4585

+ use  ` o.UsePathStyle = true` s3client option
